src/scripts/taco-rewards/taco-rewards.js

```diff
@@ -1,12 +1,162 @@
+        resultsPerQuery: RESULTS_PER_QUERY,
+        lastId: lastId,
+      })
+      .toPromise()
+
+    if (queryResult.error) {
+      console.error(
+        "ERROR: No TACo Auth history retrieved\n" + queryResult.error
+      )
+      return []
+    }
+
+    data = queryResult.data.appAuthHistories
+    if (data.length > 0) {
+      rawHistory = rawHistory.concat(queryResult.data.appAuthHistories)
+      lastId = data[data.length - 1].id
+    }
+  } while (data.length > 0)
+
+  // Let's convert this array in a dictionary
+  const authHistory = rawHistory.reduce((acc, curr) => {
+    const stakingProvider = curr.appAuthorization.stake.id
+    const historyElem = {
+      amount: curr.amount,
+      timestamp: curr.timestamp,
+      eventType: curr.eventType,
+      beneficiary: curr.appAuthorization.stake.beneficiary,
+    }
+    acc[stakingProvider] = acc[stakingProvider] || []
+    acc[stakingProvider].push(historyElem)
+    return acc
+  }, {})
+
+  return authHistory
+}
+
+// Return all the TACo commitments made
+async function getTACoCommitments(graphqlClient) {
+  const tacoCommitsQuery = gql`
+    query TacoCommitments {
+      tacoCommitments(first: 1000) {
+        id
+        endCommitment
+        duration
+      }
+    }
+  `
+
+  const queryResult = await graphqlClient.query(tacoCommitsQuery).toPromise()
+  if (queryResult.error || queryResult.data.length === 0) {
+    console.error("ERROR: No TACo commitments retrieved\n" + queryResult.error)
+    return []
+  }
+
+  const commitments = queryResult.data.tacoCommitments.reduce((acc, curr) => {
+    const stakingProvider = curr.id
+    const commitElement = {
+      endCommitment: curr.endCommitment,
+      duration: curr.duration,
+    }
+    acc[stakingProvider] = commitElement
+    return acc
+  }, {})
+
+  return commitments
+}
+
+// Return all the rewards earned by stakes that did the TACo commitment
+async function getCommitmentBonus(subgraphClient) {
+  const commitmentDeadline = 1705363199
+  const tacoCommits = await getTACoCommitments(subgraphClient)
+  const tacoAuthHistories = await getTACoAuthHistoryUntil(
+    subgraphClient,
+    commitmentDeadline
+  )
+
+  const rewards = {}
+  Object.keys(tacoCommits).map((stProv) => {
+    const stProvCheckSum = ethers.utils.getAddress(stProv)
+    const beneficiaryCheckSum = ethers.utils.getAddress(
+      tacoAuthHistories[stProv][0].beneficiary
+    )
+    const duration = tacoCommits[stProv].duration
+    const authorized = tacoAuthHistories[stProv].reduce((prev, elem) => {
+      const authorized = BigNumber(elem.amount)
+      return authorized.gt(prev) ? authorized : prev
+    }, BigNumber(0))
+
+    let reward = BigNumber(0)
+    if (duration === 3) {
+      reward = authorized.times(0.005)
+    } else if (duration === 6) {
+      reward = authorized.times(0.01)
+    } else if (duration === 12) {
+      reward = authorized.times(0.02)
+    } else if (duration === 18) {
+      reward = authorized.times(0.03)
+    }
```

Comment:
two questions here:
* How is that duration is measured in months? Where's is this calculated?
* I'm always a bit wary of float number arithmetic. How about you e.g., multiply by 3 and then integer division by 100? Later, you don't need to round it with `toFixed(0)`. I don't think it would make a big difference anyway, but it's good practice that we use integer arithmetic as much as possible.

Reply:
> * How is that duration is measured in months? Where's is this calculated?

This duration is provided by the subgraph here: 

https://github.com/threshold-network/staking-subgraphs/blob/9797e9fd44189bc0c9b0b7a02d2a650045e48f84/subgraphs/mainnet/src/taco-application.ts#L29-L56

> I'm always a bit wary of float number arithmetic. How about you e.g., multiply by 3 and then integer division by 100? Later, you don't need to round it with toFixed(0). I don't think it would make a big difference anyway, but it's good practice that we use integer arithmetic as much as possible.

You're right. But it seems like using 

```
let reward = BigNumber(0)
    if (duration === 3) {
      reward = authorized.times(5).div(1000)
    } else if (duration === 6) {
      reward = authorized.times(10).div(1000)
    } else if (duration === 12) {
      reward = authorized.times(20).div(1000)
    } else if (duration === 18) {
      reward = authorized.times(30).div(1000)
    }
```

doesn't make any difference. Maybe it's because we are using `bignumber.js` library that works fine with floats numbers.